src/plugins/ui_stateful/selection.ts

```diff
@@ -602,6 +604,43 @@ export class GridSelectionPlugin extends UIPlugin {
       sheetName: cmd.sheetName,
       elements: toRemove,
     });
+    this.applyFigureUpdates(figureUpdateCmds);
+  }
+
+  private getFiguresUpdates(cmd: MoveColumnsRowsCommand): UpdateFigureCommand[] {
+    const cmds: UpdateFigureCommand[] = [];
+    const moveDict = {};
+    const base = cmd.elements[0] < cmd.base ? cmd.base - cmd.elements.length + 1 : cmd.base;
+    for (let index = 0; index < cmd.elements.length; index++) {
+      moveDict[cmd.elements[index]] = base + index;
+    }
+    for (const figure of this.getters.getFigures(cmd.sheetId)) {
+      if (cmd.dimension == "COL" && figure.col in moveDict) {
+        cmds.push({
+          type: "UPDATE_FIGURE",
+          sheetId: cmd.sheetId,
+          figureId: figure.id,
+          col: moveDict[figure.col],
+          row: figure.row,
+        });
+      }
+      if (cmd.dimension == "ROW" && figure.row in moveDict) {
+        cmds.push({
+          type: "UPDATE_FIGURE",
+          sheetId: cmd.sheetId,
+          figureId: figure.id,
+          row: moveDict[figure.row],
+          col: figure.row,
```

Comment:
```suggestion
          col: figure.col,
```
maybe a small test ^^